coredns/resolver/endpoint_slice.go

```diff
@@ -153,7 +154,17 @@ func (i *Interface) putHeadlessEndpointSlice(key, clusterID string, endpointSlic
 			continue
 		}
 
-		var records []DNSRecord
+		var (
+			records  []DNSRecord
+			hostname string
+		)
+
+		switch {
+		case endpoint.Hostname != nil && *endpoint.Hostname != "":
+			hostname = *endpoint.Hostname
+		case endpoint.TargetRef != nil && strings.ToLower((*endpoint.TargetRef).Kind) == "pod":
```

Comment:
@vthapar @aswinsuryan I assume we only want to interpret for TargetRef as a host name if it's a Pod?

Reply:
Yes